ccd/models/lasso.py

```diff
@@ -29,17 +29,26 @@ def coefficient_matrix(dates, avg_days_yr, num_coefficients):
 
     matrix = np.zeros(shape=(len(dates), 7), order='F')
 
+    # lookup optimizations
+    # Before optimization - 12.53% of total runtime
+    # After optimization  - 10.57% of total runtime
+    w12 = w * dates
```

Comment:
Doesn't matter too much, just a point of clarity.
What could help speed things up is if w34 and w56 were moved under their respective conditionals. This would mainly help where our input data set was lighting up "initialization" a lot.